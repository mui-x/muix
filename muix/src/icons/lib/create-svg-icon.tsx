import React from 'react';
import SvgIcon from 'reactxx-mui-web/SvgIcon/SvgIcon';

const createSvgIcon = (data: string, displayName: string, isMdi:boolean) => {
    const Icon: React.SFC = props => <SvgIcon {...props} >
        <path d={data} />
    </SvgIcon >

    const res: any = Icon
    res.displayName = displayName;
    res.muiName = 'SvgIcon';

    return res;
}

export default createSvgIcon;