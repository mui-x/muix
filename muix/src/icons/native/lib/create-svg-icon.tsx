import React from 'react';

const createSvgIcon = (data: string, displayName: string, isMdi:boolean) => {
    return null
}

export default createSvgIcon;