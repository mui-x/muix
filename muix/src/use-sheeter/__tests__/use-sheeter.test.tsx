/** @jsx platform.createElement */

import { platform } from "reactxx-sheeter"

import { initPlatform, render } from "./init-platform.t"
import { compCreator } from "./lib.t"

describe("USE SHEETER", () => {

  const doTest = (isWeb: boolean) => {
    beforeEach(() => {
      initPlatform(isWeb, { dataTraceFlag: 'short' })
      platform.getDefaultTheme = () => ({
        primaryColor: 'gray',
        p1Prop: 'themeProp'
      })
    })

    it('01: missing configs', () => {
      const Comp = compCreator(null)
      const wrapper = render(<Comp css={{margin: 20}}/>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('02: sheet missing', () => {
      const Comp = compCreator({}, null)
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })
    it('03: just div', () => {
      const Comp = compCreator({ defaultSheet: { root: {} } }, null)
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('04: with sheet', () => {
      const Comp = compCreator({ defaultSheet: { root: { color: 'red' } } }, null)
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('05: with sheet override', () => {
      const Comp = compCreator(
        { defaultSheet: { root: { color: 'red' } } },
        { overrideSheet: { root: { color: 'green' } } }
      )
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('06: with classes override', () => {
      const Comp = compCreator(
        { defaultSheet: { root: { color: 'red' } } },
        { overrideSheet: { root: { color: 'green' } } }
      )
      const wrapper = render(<Comp classes={{ root: { color: 'blue' } }} />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('07: with css override', () => {
      const Comp = compCreator(
        { defaultSheet: { root: { color: 'red' } } },
        { overrideSheet: { root: { color: 'green' } } }
      )
      const wrapper = render(<Comp classes={{ root: { color: 'blue' } }} css={{ color: 'yellow' }} />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('08: with style override', () => {
      const Comp = compCreator(
        { defaultSheet: { root: { color: 'red' } } },
        { overrideSheet: { root: { color: 'green' } } }
      )
      const wrapper = render(<Comp classes={{ root: { color: 'blue' } }} css={{ color: 'yellow' }} styles={{ color: 'maroon' }} />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('09: theme, with sheet', () => {
      const Comp = compCreator({ defaultSheet: theme => ({ root: { color: theme.primaryColor } }) }, null)
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('10: theme, with sheet override', () => {
      const Comp = compCreator(
        { defaultSheet: { root: { color: 'red' } } },
        { overrideSheet: theme => ({ root: { color: theme.primaryColor } }) }
      )
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('11: theme, with classes override', () => {
      const Comp = compCreator(
        { defaultSheet: { root: { color: 'red' } } },
        { overrideSheet: { root: { color: 'green' } } }
      )
      const wrapper = render(<Comp classes={theme => ({ root: { color: theme.primaryColor } })} />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('12: theme, with classNameX override', () => {
      const Comp = compCreator(
        { defaultSheet: { root: { color: 'red' } } },
        { overrideSheet: { root: { color: 'green' } } }
      )
      const wrapper = render(<Comp classes={{ root: { color: 'blue' } }} css={theme => ({ color: theme.primaryColor })} />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('13: theme, with styleX override', () => {
      const Comp = compCreator(
        { defaultSheet: { root: { color: 'red' } } },
        { overrideSheet: { root: { color: 'green' } } }
      )
      const wrapper = render(<Comp classes={{ root: { color: 'blue' } }} css={{ color: 'yellow' }} styles={theme => ({ color: theme.primaryColor })} />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })


    it('14: default props', () => {
      const Comp = compCreator(
        {
          defaultSheet: { root: {} },
          defaultProps: { p1: 'defaultProp' }
        },
        null
      )
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('15: override props', () => {
      const Comp = compCreator(
        {
          defaultSheet: { root: {} },
          defaultProps: { p1: 'defaultProp' }
        },
        { overrideProps: { p1: 'overrideProp' } }
      )
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('16: props', () => {
      const Comp = compCreator(
        {
          defaultSheet: { root: {} },
          defaultProps: { p1: 'defaultProp' }
        },
        { overrideProps: { p1: 'overrideProp' } }
      )
      const wrapper = render(<Comp p1='componentProp' />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('17: theme, default props', () => {
      const Comp = compCreator(
        {
          defaultSheet: { root: {} },
          defaultProps: { p1: 'defaultProp', themedProps: theme => ({ p1: theme.p1Prop }) }
        },
        null
      )
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('18: theme, override props', () => {
      const Comp = compCreator(
        {
          defaultSheet: { root: {} },
          defaultProps: { p1: 'defaultProp' }
        },
        { overrideProps: { p1: 'overrideProp', themedProps: theme => ({ p1: theme.p1Prop }) } }
      )
      const wrapper = render(<Comp />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('19: theme, props', () => {
      const Comp = compCreator(
        {
          defaultSheet: { root: {} },
          defaultProps: { p1: 'defaultProp' }
        },
        { overrideProps: { p1: 'overrideProp' } }
      )
      const wrapper = render(<Comp p1='componentProp' themedProps={theme => ({ p1: theme.p1Prop })} />)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('20: style', () => {
      const Comp = compCreator({ defaultSheet: { root: {} } })
      const wrapper = render(<Comp styles={{color: 'red', margin: 5, $web: {margin: 10}, $native: {margin: 20}}}/>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('21: style with classes', () => {
      const Comp = compCreator({ defaultSheet: { root: { color: 'red'} } })
      const wrapper = render(<Comp css={{margin: 20}} styles={{color: 'blue', $web: {margin: 30}, $native: {margin: 40}}}/>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

    it('22: user config only', () => {
      const Comp = compCreator(null, { overrideSheet: { root: { color: 'red'} } })
      const wrapper = render(<Comp/>)
      expect(wrapper.container).toMatchSnapshot()
      wrapper.unmount()
    })

  }

  describe("## NATIVE ##", () => doTest(false))
  describe("## WEB ##", () => doTest(true))

})

