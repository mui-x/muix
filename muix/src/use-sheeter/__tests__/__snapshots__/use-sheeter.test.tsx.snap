// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USE SHEETER ## NATIVE ## 01: missing configs 1`] = `
<div>
  <div
    style="margin: 20px;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 02: sheet missing 1`] = `
<div>
  <div>
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 03: just div 1`] = `
<div>
  <div>
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 04: with sheet 1`] = `
<div>
  <div
    style="color: red;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 05: with sheet override 1`] = `
<div>
  <div
    style="color: green;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 06: with classes override 1`] = `
<div>
  <div
    style="color: blue;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 07: with css override 1`] = `
<div>
  <div
    style="color: yellow;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 08: with style override 1`] = `
<div>
  <div
    style="color: maroon;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 09: theme, with sheet 1`] = `
<div>
  <div
    style="color: gray;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 10: theme, with sheet override 1`] = `
<div>
  <div
    style="color: gray;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 11: theme, with classes override 1`] = `
<div>
  <div
    style="color: gray;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 12: theme, with classNameX override 1`] = `
<div>
  <div
    style="color: gray;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 13: theme, with styleX override 1`] = `
<div>
  <div
    style="color: gray;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 14: default props 1`] = `
<div>
  <div>
    defaultProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 15: override props 1`] = `
<div>
  <div>
    overrideProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 16: props 1`] = `
<div>
  <div>
    componentProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 17: theme, default props 1`] = `
<div>
  <div>
    themeProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 18: theme, override props 1`] = `
<div>
  <div>
    themeProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 19: theme, props 1`] = `
<div>
  <div>
    themeProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 20: style 1`] = `
<div>
  <div>
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 21: style with classes 1`] = `
<div>
  <div
    style="color: blue; margin: 40px;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## NATIVE ## 22: user config only 1`] = `
<div>
  <div
    style="color: red;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 01: missing configs 1`] = `
<div>
  <div
    class="a"
    data-trace="
.a { margin:20px }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 02: sheet missing 1`] = `
<div>
  <div>
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 03: just div 1`] = `
<div>
  <div>
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 04: with sheet 1`] = `
<div>
  <div
    class="a"
    data-trace="
.a { color:red }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 05: with sheet override 1`] = `
<div>
  <div
    class="b"
    data-trace="
.b { color:green }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 06: with classes override 1`] = `
<div>
  <div
    class="c"
    data-trace="
.c { color:blue }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 07: with css override 1`] = `
<div>
  <div
    class="d"
    data-trace="
.d { color:yellow }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 08: with style override 1`] = `
<div>
  <div
    class="d"
    data-trace="
.d { color:yellow }"
    style="color: maroon;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 09: theme, with sheet 1`] = `
<div>
  <div
    class="a"
    data-trace="
.a { color:gray }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 10: theme, with sheet override 1`] = `
<div>
  <div
    class="b"
    data-trace="
.b { color:gray }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 11: theme, with classes override 1`] = `
<div>
  <div
    class="c"
    data-trace="
.c { color:gray }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 12: theme, with classNameX override 1`] = `
<div>
  <div
    class="d"
    data-trace="
.d { color:gray }"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 13: theme, with styleX override 1`] = `
<div>
  <div
    class="d"
    data-trace="
.d { color:yellow }"
    style="color: gray;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 14: default props 1`] = `
<div>
  <div>
    defaultProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 15: override props 1`] = `
<div>
  <div>
    overrideProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 16: props 1`] = `
<div>
  <div>
    componentProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 17: theme, default props 1`] = `
<div>
  <div>
    themeProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 18: theme, override props 1`] = `
<div>
  <div>
    themeProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 19: theme, props 1`] = `
<div>
  <div>
    themeProp: 1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 20: style 1`] = `
<div>
  <div
    style="color: red; margin: 10px;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 21: style with classes 1`] = `
<div>
  <div
    class="b a"
    data-trace="
.b { margin:20px }
.a { color:red }"
    style="color: blue; margin: 30px;"
  >
    1
  </div>
</div>
`;

exports[`USE SHEETER ## WEB ## 22: user config only 1`] = `
<div>
  <div
    class="a"
    data-trace="
.a { color:red }"
  >
    1
  </div>
</div>
`;
